jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ExtraDirectoriesParameters.java

```diff
@@ -106,15 +125,15 @@ public void setPaths(Object paths) {
    * @return the permissions map from path on container to file permissions
    */
   @Input
-  public Map<String, String> getPermissions() {
+  public MapProperty<String, String> getPermissions() {
     String property = System.getProperty(PropertyNames.EXTRA_DIRECTORIES_PERMISSIONS);
     if (property != null) {
-      return ConfigurationPropertyValidator.parseMapProperty(property);
+      Map<String, String> parsedPermissions =
+          ConfigurationPropertyValidator.parseMapProperty(property);
+      if (!parsedPermissions.equals(permissions.get())) {
```

Comment:
Does this check even matter? There is no harm in setting permissions to an equivalent object.

Reply:
True! I saw that both PR #3242 and #3034 both included this check in the getter, and my understanding was that we wanted to avoid setting a property multiple times here?